test/cpp/jit/test_gpu.cpp

```diff
@@ -5125,6 +5125,221 @@ TEST(NVFuserTest, FusionBCastAfterReduce_CUDA) {
   TORCH_CHECK(t5.allclose(outputs[0], 1e-5, 1e-5));
 }
 
+TEST(NVFuserTest, FusionOutputBroadcast_CUDA) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  TensorView* tv0 = makeConcreteTensor({2, 3});
+  fusion.addInput(tv0);
+
+  TensorView* tv1 = broadcast(tv0, {true, false, true, false, true});
+
+  fusion.addOutput(tv1);
+
+  const auto options =
+      at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+
+  at::Tensor input = at::randn({2, 3}, options);
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion);
+
+  auto outputs = fe.runFusion({input});
+  auto aten_output = input;
+
+  TORCH_CHECK(
+      aten_output.allclose(outputs[0].squeeze(), 1e-04, 1e-04),
+      "Error of: ",
+      aten_output.sub(outputs[0].squeeze()).abs().max());
+}
+
+TEST(NVFuserTest, FusionReductionKeepDimBasic_CUDA) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  TensorView* tv0 = makeConcreteTensor({2, 3, 4, 5, 6});
+  fusion.addInput(tv0);
+
+  TensorView* tv1 = sum(tv0, {0, 2, 4}, /*keep_dim=*/true);
+
+  fusion.addOutput(tv1);
+
+  const auto options =
+      at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+
+  at::Tensor input = at::randn({2, 3, 4, 5, 6}, options);
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion);
+
+  auto outputs = fe.runFusion({input});
+  auto aten_output = input.sum({0, 2, 4});
+
+  TORCH_CHECK(
+      aten_output.allclose(outputs[0].squeeze(), 1e-04, 1e-04),
+      "Error of: ",
+      aten_output.sub(outputs[0].squeeze()).abs().max());
+}
+
+TEST(NVFuserTest, FusionReductionKeepDimScheduler_CUDA) {
+  constexpr int bid_x = 80;
+  constexpr int tid_x = 4096;
+  constexpr int red_dim = 1;
+
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  // Set up your input tensor views
+  TensorView* tv0 = makeConcreteTensor({bid_x, tid_x});
+  fusion.addInput(tv0);
+
+  TensorView* tv1 = reductionOp(
+      BinaryOpType::Add, {red_dim}, new Float(0), tv0, /*keep_dim=*/true);
+
+  TensorView* red_tv = fusion.origin(tv1)->inputs()[0]->as<TensorView>();
+
+  fusion.addOutput(tv1);
+
+  const auto options =
+      at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor input = at::randn({bid_x, tid_x}, options);
+
+  // Apply reduction heuristic
+  auto reduction_params = getReductionHeuristics(&fusion, {input}, red_tv);
+  TORCH_CHECK(reduction_params, "Reduction schedule was not generated!");
+  scheduleReduction(&fusion, reduction_params.value(), red_tv, {tv1});
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion);
+
+  auto outputs = fe.runFusion({input}, reduction_params.value().lparams);
+  auto aten_output = input.sum({red_dim});
+
+  TORCH_CHECK(
+      aten_output.allclose(outputs[0].squeeze(), 1e-04, 1e-04),
+      "Error of: ",
+      aten_output.sub(outputs[0].squeeze()).abs().max());
+}
+
+TEST(NVFuserTest, FusionSumKeepDimScheduler_CUDA) {
```

Comment:
Do we need another test to check sum explicitly?

Reply:
It was just quick API test but I guess it wasn't necessary. Removed sum tests.